recipes/inkscape/wrapper.c

```diff
@@ -0,0 +1,262 @@
+        return 1;
+    }
+    child_pid = p_info.dwProcessId;
+    // wait for Python to exit
+    WaitForSingleObject(p_info.hProcess, INFINITE);
+    if (!GetExitCodeProcess(p_info.hProcess, &return_value))
+    {
+        fprintf(stderr, "failed to get exit code from process.\n");
+        return 0;
+    }
+    return return_value;
+}
+
+
+char* join_executable_and_args(char *executable, char **args, int argc)
+{
+    /*
+     * distribute-issue207
+     * CreateProcess needs a long string of the executable and command-line arguments,
+     * so we need to convert it from the args that was built
+     */
+    int len, counter;
+    char* cmdline;
+
+    len = strlen(executable) + 2;
+    for (counter=1; counter<argc; counter++)
+    {
+        len += strlen(args[counter]) + 1;
+    }
+
+    cmdline = (char *) calloc(len, sizeof(char));
+    sprintf(cmdline, "%s", executable);
+    len=strlen(executable);
+    for (counter=1; counter<argc; counter++) {
+        sprintf(cmdline+len, " %s", args[counter]);
+        len += strlen(args[counter]) + 1;
+    }
+    return cmdline;
+}
+
+
+int run(int argc, char **argv, int is_gui)
+{
+    char path[MAX_PATH], newpath[MAX_PATH];
+    char **newargs, **newargsp; /* argument array for exec */
+    char *fn, *end, *ext;     /* working pointers for string manipulation */
+    char *cmdline;
+    int i;              /* loop counter */
+
+    /* compute script name from our .exe name*/
+    GetModuleFileNameA(NULL, path, sizeof(path));
+
+
+    fn = path + strlen(path);
+    /*
+    walk from the end to the last slash -> fn is the name of this wrapper.
+    including the extension
+    */
+    while (fn > path && *fn != '\\') {
+        fn--;
+    }
+    fn++;
+    end = fn - 2;
+    while (end > path && *end != '\\'){
+        end--;
+    }
+    *end = '\0';
+
+    ext = fn + strlen(fn);
+    while (ext > path && *ext != '.') {
+        ext--;
+    }
+    if (ext > fn) {
+        *ext = '\0';
+    }
+
+    strcpy(newpath, path);
+    strcat(newpath, RELATIVE_PATH);
+    strcat(newpath, fn);
+    strcat(newpath, EXEC_EXTENSION);
+
+#if DEBUG
+    printf("fn ==%s==\n", fn);
+    printf("==%s==\n", newpath);
+#endif
+    /* Argument array needs to be argc, plus 1 for null sentinel */
+    newargs = (char **) calloc(argc + 1, sizeof(char *));
+    newargsp = newargs;
+
+    *newargsp++ = quoted(newpath);
+    for (i = 1; i < argc; i++)
+        *newargsp++ = quoted(argv[i]);
+
+    *newargsp++ = NULL;
+
+#if DEBUG
+    for (i = 0; i <= argc; i++)
+        printf("- %s\n", newargs[i]);
+    printf("argc=%d\n", argc);
+#endif
+
+    if (is_gui) {
+        /* Use exec, we don't need to wait for the GUI to finish */
+        execv(newpath, (char * const *) (newargs));
+        return fail("Could not exec %s", newpath); /* shouldn't get here! */
+    }
+
+    /*
+     * distribute-issue207: using CreateProcessA instead of spawnv
+     */
+    cmdline = join_executable_and_args(newpath, newargs, argc);
```

Comment:
I kept that as is for the time being (both because I guess that the string is freed on exit and also because I've not yet a clue how to rewrite it to do that)...